detect_secrets/core/scan.py

```diff
@@ -326,14 +328,28 @@ def _scan_line(
     filename: str,
     line: str,
     line_number: int,
+    **kwargs: Any,
 ) -> Generator[PotentialSecret, None, None]:
     # NOTE: We don't apply filter functions here yet, because we don't have any filters
     # that operate on (filename, line, plugin) without `secret`
     try:
-        secrets = plugin.analyze_line(filename=filename, line=line, line_number=line_number)
+        analyze_line = plugin.__class__.analyze_line
+        if not hasattr(analyze_line, 'injectable_variables'):
+            analyze_line.injectable_variables = set(        # type: ignore
+                get_injectable_variables(plugin.analyze_line),
+            )
+            analyze_line.path = f'{plugin.__class__.__name__}.analyze_line'     # type: ignore
     except AttributeError:
```

Comment:
does this `except` need to be after `secrets = inject_variables_into_function(...)`?